examples/05_deduplication.py

```diff
@@ -131,48 +118,71 @@
+# create a table that maps original to corrected categories
 translation_table = pd.Series(deduplicated_data, index=duplicated_names)
 
 # remove duplicates in the original data
 translation_table = translation_table[~translation_table.index.duplicated(keep="first")]
 
+translation_table.head()
+
+###############################################################################
+# In this table, we have the category name on the left,
+# and the cluster it was translated to on the right.
+# If we want to adapt the translation table, we can
+# modify it manually.
+
 ###############################################################################
-# Since the number of correct spellings will likely be much smaller than the
-# number of original categories, we can print the estimated cluster and their
-# most common exemplars (the guessed correct spelling):
+# Visualizing string pair-wise distance between names
+# ---------------------------------------------------
+#
+# Below, we use a heatmap to visualize the pairwise-distance between medication
+# names. A darker color means that two medication names are closer together
+# (i.e. more similar), a lighter color means a larger distance.
+#
 
+from skrub import compute_ngram_distance
+from scipy.spatial.distance import squareform
 
-def print_corrections(spell_correct):
-    correct = np.unique(spell_correct.values)
-    for c in correct:
-        print(
-            f"Guessed correct spelling: {c!r} for "
-            f"{spell_correct[spell_correct==c].index.values}"
-        )
+ngram_distances = compute_ngram_distance(unique_examples)
+square_distances = squareform(ngram_distances)
 
+import seaborn as sns
 
-print_corrections(translation_table)
+fig, ax = plt.subplots(figsize=(14, 12))
+sns.heatmap(
+    square_distances, yticklabels=ex_series.index, xticklabels=ex_series.index, ax=ax
+)
+plt.show()
 
 ###############################################################################
-# In case we want to adapt the translation table post-hoc we can easily
-# modified it manually and apply it, for instance modifying the
-# correspondance for the last entry as such:
+# We have three clusters appearing - the original medication
+# names and their misspellings that form a cluster around them.
 
-translation_table.iloc[-1] = "Completely new category"
-new_deduplicated_names = translation_table[duplicated_names]
-assert (new_deduplicated_names == "Completely new category").sum() > 0
+###############################################################################
+# Conclusion
+# ----------
+#
+# In this example, we have seen how to use the |deduplicate| function to
+# automatically detect and correct misspelled category names.
+#
+# Note that deduplication is especially useful when we either
+# know our ground truth (e.g. the original medication names),
+# or when we cannot use the similarity across string directly
+# in a machine learning task.
+# Otherwise, it is better to use encoding methods such as |Gap|
```

Comment:
Can we explain why Gap and MinHash encoders are better than deduplication, and in which scenarios?

Reply:
Modified, check if this version is ok for you !

Reply:
Yes, thanks! I suggested changes for few typos